### PR TITLE
QCAS Surveys

### PR DESCRIPTION
**Pull request by MebinAbraham**
### What is the context of this PR?
***New QCAS surveys*** 
*3 form types*:
- 0018 - No minerals section 
- 0019 - No minerals + NI theme
- 0020 - Includes minerals section

Survey info and discussions can be found on [Trello](https://trello.com/c/RRhO27ZU/2278-build-qcas-survey).

### How to review 
Make sure all content and structure is as expected.
[Docs for structure](https://docs.google.com/spreadsheets/d/1DSLtru61W7F27K7tWZjsfMHJFN5kpcpS4qNsHy30LEY/edit#gid=0)

> Note: Introduction might change slightly, waiting on Jimi to confirm.

Changes since definition:
- Confirmation question needed for `Improvement and construction work`.
- Answer guidance to be only on the first question.
- `Improvement and construction work` not included in the Calculated playback.

*Quick launch:*
[0018](https://preprod-new-surveys-launch.eq.ons.digital/quick-launch?url=https://raw.githubusercontent.com/ONSdigital/eq-survey-runner/master/data/en/qcas_0018.json)
[0019](https://preprod-new-surveys-launch.eq.ons.digital/quick-launch?url=https://raw.githubusercontent.com/ONSdigital/eq-survey-runner/master/data/en/qcas_0019.json)
[0020](https://preprod-new-surveys-launch.eq.ons.digital/quick-launch?url=https://raw.githubusercontent.com/ONSdigital/eq-survey-runner/master/data/en/qcas_0020.json)

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: data/en/qcas_0018.json [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+                                            "Work contracted to constructors or arranged through agents or developers.",
+                                            "Non-residential buildings, for example, public monuments, commercial, industrial, educational and health buildings.",
+                                            "Fittings and installations such as lifts, electrics, plumbing, heating and ventilation systems and water supply.",
+                                            "Construction of structures such as oil wells, platforms, mines, pipes, power lines, transport infrastructure, wind farms, turbines and steel frameworks.",
+                                            "Site improvements and other land preparation for own use, for example, levelling, clearance and draining of marshes.",
+                                            "Transfer costs, stamp duties and taxes payable to acquire the asset.",
+                                            "All professional charges such as legal costs, agents&apos; commission, and architects&apos;, engineers&apos;, and surveyors&apos; fees.",
```
Comment (dwyeradam) [inline data/en/qcas_0018.json]: Replace all instances of `&apos;` with smart quotes `’`.

**Review by dwyeradam — CHANGES_REQUESTED** [review]
"Don't be dumb"
“You’re smart!”

File: data/en/qcas_0018.json [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+                            "type": "General",
+                            "answers": [{
+                                    "id": "machinery-equipment-acquisitions-answer-1",
+                                    "mandatory": true,
+                                    "type": "Currency",
+                                    "q_code": "688",
+                                    "label": "Value of acquisitions for transport assets and equipment",
+                                    "decimal_places": 2,
+                                    "currency": "GBP"
+                                },
```
Comment (dwyeradam) [inline data/en/qcas_0018.json:373]: No "description": "Figures can be entered rounded (eg 56,000) or in full (eg 56,234.33). Do not enter ‘56’ for £56,000." shouldn't there be for all of these?

File: data/en/qcas_0019.json [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+                                        "value": "Change of business structure, merger, or takeover",
+                                        "q_code": "146f"
+                                    },
+                                    {
+                                        "label": "One off or unusual investment",
+                                        "value": "One off or unusual investment",
+                                        "q_code": "146g"
+                                    },
+                                    {
+                                        "label": "Introduction/removal of new legislation/incentive",
```
Comment (bitdivision) [inline data/en/qcas_0019.json]: This is inconsistent with the spreadsheet.

File: data/en/qcas_0018.json [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+                        }]
+                    },
+                    {
+                        "id": "significant-changes-reason-block-2",
+                        "title": "Significant changes",
+                        "type": "Question",
+                        "questions": [{
+                            "id": "significant-changes-reason-question-2",
+                            "title": "Please describe the changes in total investment for {{ metadata['trad_as_or_ru_name'] }} in more detail.",
+                            "description": "We rely on your commentary to &#x2018;tell the story&#x2019; behind changes in figures. By commenting here it will reduce the need for us to call you.",
```
Comment (dwyeradam) [inline data/en/qcas_0018.json]: Don't forget to convert `&#x2018;` and `&#x2019;` as well.
‘significant change’

**Review by dwyeradam — CHANGES_REQUESTED** [review]

File: data/en/qcas_0018.json [hunk 5]
```diff
@@ -0,0 +1,739 @@
...
+                        "routing_rules": [{
+                                "goto": {
+                                    "block": "significant-changes-reason-block-1",
+                                    "when": [{
+                                            "id": "answer1938",
+                                            "condition": "not equals",
+                                            "value": "No"
+                                        },
+                                        {
+                                            "id": "answer1938",
```
Comment (AndrewTorrance) [inline data/en/qcas_0018.json]: set check not needed as answer1938 is mandatory

File: data/en/qcas_0019.json (hunk 3 above)
Comment (dwyeradam) [reply data/en/qcas_0019.json]: This has come up before. On other surveys (e.g. 1_0102.json) we have spaces around `/`. We should be consistent across surveys. 
Introduction / removal of new legislation / incentive

**Comment by bitdivision** [comment]
Have reviewed against spreadsheet / docs. No major differences.

File: data/en/qcas_0018.json [hunk 6]
```diff
@@ -0,0 +1,739 @@
...
+                        "routing_rules": [{
+                                "goto": {
+                                    "block": "improvements-construction-block",
+                                    "when": [{
+                                            "id": "reporting-period-answer",
+                                            "condition": "equals",
+                                            "value": "Yes"
+                                        },
+                                        {
+                                            "id": "reporting-period-answer",
```
Comment (AndrewTorrance) [inline data/en/qcas_0018.json]: set check not needed , reporting period answer is mandatory

File: data/en/qcas_0018.json [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+    "eq_id": "qcas",
+    "form_type": "0018",
+    "mime_type": "application/json/ons/eq",
+    "schema_version": "0.0.1",
+    "data_version": "0.0.1",
+    "survey_id": "019",
+    "title": "Quarterly Acquisitions and Disposals of Capital Assets Survey",
+    "theme": "default",
+    "legal_basis": "StatisticsOfTradeAct",
+    "navigation": {
```
Comment (dcdarrell9) [inline data/en/qcas_0018.json]: Do we need this? Does it default to False anyway?

File: data/en/qcas_0019.json [hunk 8]
```diff
@@ -0,0 +1,739 @@
...
+                        "routing_rules": [{
+                                "goto": {
+                                    "block": "improvements-construction-block",
+                                    "when": [{
+                                            "id": "reporting-period-answer",
+                                            "condition": "equals",
+                                            "value": "Yes"
+                                        },
+                                        {
+                                            "id": "reporting-period-answer",
```
Comment (AndrewTorrance) [inline data/en/qcas_0019.json]: set check not needed - mandatory answer

File: data/en/qcas_0019.json [hunk 9]
```diff
@@ -0,0 +1,739 @@
...
+                        }],
+                        "routing_rules": [{
+                                "goto": {
+                                    "block": "significant-changes-reason-block-1",
+                                    "when": [{
+                                            "id": "answer1938",
+                                            "condition": "not equals",
+                                            "value": "No"
+                                        },
+                                        {
```
Comment (AndrewTorrance) [inline data/en/qcas_0019.json]: set check redundant as answer mandatory

File: data/en/qcas_0020.json [hunk 10]
```diff
@@ -0,0 +1,774 @@
...
+                                "goto": {
+                                    "block": "improvements-construction-block",
+                                    "when": [{
+                                            "id": "reporting-period-answer",
+                                            "condition": "equals",
+                                            "value": "Yes"
+                                        },
+                                        {
+                                            "id": "reporting-period-answer",
+                                            "condition": "set"
```
Comment (AndrewTorrance) [inline data/en/qcas_0020.json]: Not needed - answer mandatory

File: data/en/qcas_0020.json [hunk 11]
```diff
@@ -0,0 +1,774 @@
...
+                        "routing_rules": [{
+                                "goto": {
+                                    "block": "significant-changes-reason-block-1",
+                                    "when": [{
+                                            "id": "answer1938",
+                                            "condition": "not equals",
+                                            "value": "No"
+                                        },
+                                        {
+                                            "id": "answer1938",
```
Comment (AndrewTorrance) [inline data/en/qcas_0020.json]: Not needed , answer mandatory

**Review by AndrewTorrance — CHANGES_REQUESTED** [review]
a few instances of routing rules checking for value set on mandatory answers

File: data/en/qcas_0018.json (hunk 2 above)
Comment (MebinAbraham) [reply data/en/qcas_0018.json:373]: Apparently No, Will said to remove them for all but the first question after checking with Jordan.

File: data/en/qcas_0018.json [hunk 12]
```diff
@@ -0,0 +1,739 @@
...
+                            "title": "Did any significant changes occur to the total investment for {{ metadata['trad_as_or_ru_name'] }}?",
+                            "definitions": [{
+                                "title": "What constitutes a significant change?",
+                                "content": [{
+                                    "description": "What constitutes a ‘significant change’ is dependent on your own interpretation in relation to {{ metadata['trad_as_or_ru_name'] }}’s figures from the previous reporting period and the same reporting period last year."
+                                }]
+                            }],
+                            "type": "General",
+                            "answers": [{
+                                "id": "answer1938",
```
Comment (dcdarrell9) [inline data/en/qcas_0018.json]: is this right?

File: data/en/qcas_0018.json (hunk 7 above)
Comment (MebinAbraham) [reply data/en/qcas_0018.json]: Was waiting on business to make their mind up.

File: data/en/qcas_0018.json (hunk 12 above)
Comment (MebinAbraham) [reply data/en/qcas_0018.json]: Its not wrong, but ill change to make it meaningful, missed this one.

**Comment by MebinAbraham** [comment]
@dwyeradam @bitdivision @dcdarrell9 @AndrewTorrance 
Can be re-reviewed. Thanks

File: data/en/qcas_0020.json [hunk 13]
```diff
@@ -0,0 +1,761 @@
...
+                                        "value": "Change of business structure, merger, or takeover",
+                                        "q_code": "146f"
+                                    },
+                                    {
+                                        "label": "One off or unusual investment",
+                                        "value": "One off or unusual investment",
+                                        "q_code": "146g"
+                                    },
+                                    {
+                                        "label": "Introduction/removal of new legislation/incentive",
```
Comment (bitdivision) [inline data/en/qcas_0020.json]: More spaces

Comment (MebinAbraham) [reply data/en/qcas_0020.json]: I swear it got lost in transmission -_-. Will update 👍

File: data/en/qcas_0018.json [hunk 14]
```diff
@@ -0,0 +1,726 @@
...
+                            }]
+                        }],
+                        "routing_rules": [{
+                                "goto": {
+                                    "block": "significant-changes-reason-block-1",
+                                    "when": [{
+                                        "id": "any-significant-changes-answer",
+                                        "condition": "not equals",
+                                        "value": "No"
+                                    }]
```
Comment (dwyeradam) [inline data/en/qcas_0018.json]: `not equals no` you mean `equals yes`. Although it's not wrong we should probably avoid the double negative.

Comment (MebinAbraham) [reply data/en/qcas_0018.json]: Missed this one. Changed the one above.

**Comment by MebinAbraham** [comment]
`DO NOT MERGE` as I am waiting on business to make a decision regarding the intro accordion being too long. They possibly will want the wording changed.

screen shot 2018-08-17 at 15 43 00

**Comment by jonnyshaw89** [comment]
🤦‍♂️

**Comment by MebinAbraham** [comment]
@bitdivision @AndrewTorrance @dwyeradam 
Added Introduction content. Good luck.

**Review by insacuri — APPROVED** [review]
Compared all 3 to the doc and tested 2 of them in preprod.  I might've missed something (cause there is quite a bit there...) but it all looks fine as far as I can tell

**Review by dcdarrell9 — APPROVED** [review]

**Comment by MebinAbraham** [comment]
From Jimi 

> The survey team came back on the wording and they don't want to remove 'total' or 'major' so guess we have to leave it like that

This is regarding the wording being too long on the intro.

**Review by dwyeradam — CHANGES_REQUESTED** [review]
We haven't add the ability to view submitted responses. I'm assuming it's required.

"view_submitted_response": {
        "enabled": true,
        "duration": 900
    },

**Review by dwyeradam — APPROVED** [review]

**Review by AndrewTorrance — APPROVED** [review]

**Pull request merged**